xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/internal/document/DocumentOverrideListener.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.internal.document;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.event.CancelableEvent;
+import org.xwiki.observation.event.Event;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.internal.event.UserUpdatingDocumentEvent;
+
+/**
+ * Cancel any save that tries to save a new document if the document already exists.
+ *
+ * @version $Id$
+ * @since 14.10.21
+ * @since 15.5.5
+ * @since 15.10.5
+ */
+@Component
+@Singleton
+@Named(DocumentOverrideListener.NAME)
+public class DocumentOverrideListener extends AbstractEventListener
+{
+    /**
+     * The unique identifier of the listener.
+     */
+    public static final String NAME = "org.xwiki.internal.document.DocumentOverrideListener";
+
+    /**
+     * The default constructor.
+     */
+    public DocumentOverrideListener()
+    {
+        super(NAME, new UserUpdatingDocumentEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        XWikiDocument document = (XWikiDocument) source;
+
+        XWikiDocument originalDocument = document.getOriginalDocument();
+
+        if (event instanceof CancelableEvent && document.isNew() && originalDocument != null
```

Comment:
UserUpdatingDocumentEvent is cancellable by definition, and you are not listening to any other event so in theory you don't need the `event instanceof CancelableEvent`.

Reply:
Okay, I've removed it.